bin/pycbc_make_skymap

```diff
@@ -448,6 +465,14 @@ if __name__ == '__main__':
                         action=MultiDetOptionAppendAction,
                         help='Lists of local frame files, e.g., '
                              'H1:/path/to/frame/file L1:/path/to/frame/file')
+    parser.add_argument('--injection-file', type=str,
+                        help='Optional path to an injection file.')
+    parser.add_argument('--fake-strain', nargs='+', type=str,
+                        action=MultiDetOptionAppendAction, metavar="IFO:FAKE_STRAIN",
+                        help="The set of fake-strains for each detector")
+    parser.add_argument('--fake-strain-seed', nargs='+', type=str,
+                        action=MultiDetOptionAppendAction, metavar="IFO:FAKE_STRAIN_SEED",
```

Comment:
The seed is an `int`, not a `str`. Also, can you try to remove the `nargs` kwarg? I think it is no longer needed now (you may have to rebase this PR on master though).

Reply:
Has this changed?

The `fake-strain-seed` option is fed into `pycbc_single_template` in the form of 'H1:1234 L1:5678'. Is this not a string (the individual H1:1234)?

When I remove `nargs='+'` I am presented the error: `The character ':' is used to distinguish the ifo and the value. It must be given exactly once for all entries`.

I believe I have rebased correctly but I might have slipped up somewhere.

Reply:
Hmm I see what you mean. You could indeed literally pass the strings `H1:1234` and `L1:5678` to `pycbc_single_template`, but note that those are two separate strings, which should be passed as two separate arguments, i.e. not a single string with a white space. If you want to go this way, then you should include `nargs='+'`, but *remove* `action=MultiDetOptionAppendAction` (since you will just ask argparse to simply pass the list of strings, instead of interpreting the H1:1234 syntax).

Reply:
When removing `action=MultiDetOptionAppendAction` it stores the `fake_strain` arguments in a list which it then can't loop over for each ifo simply by doing `fake_strain[ifo]`. I can see that a similar argument `--channel-name` creates an empty dictionary at the beginning of the code. However, because `--fake-strain` and `--fake-strain-seed` are optional arguments I don't know if this is what I want to do.

Is there another way to extract the corresponding fake strain values for each ifo?

Reply:
Yeah, sorry, I was not thinking straight in my last comment.

The initial code with `type=str` was indeed correct given that you just pass the value as a string. However, `type=int` has the small additional benefit that it makes it clear that the option is supposed to take an integer (as opposed to any random string). When you pass the value to `pycbc_single_template`, you then need to convert it back to a string like this:
```python
command.append(str(fake_strain_seed[ifo][0]))
```

Another comment I should have made before is that the action should probably be `action=MultiDetOptionAction`, and then you can switch from
```python
command.append(str(fake_strain_seed[ifo][0]))
```
to just
```python
command.append(str(fake_strain_seed[ifo]))
```
(the same goes for `--fake-strain`). The difference is that `MultiDetOptionAppendAction` allows you to pass multiple values for each detector, like `H1:123 H1:456 L1:345 L1:789`. This is appropriate for `--custom-frame-files`, but not for `--fake-strain` and `--fake-strain-seed`.

As for `nargs='+'`, it *should* work without it, but you can just leave it in otherwise.

Reply:
The need for `nargs='+'` should have been removed by #3663, by the way.

Reply:
That all makes a lot of sense!

Your explanation of the difference between the `action` arguments has cleared up a few things in my head so thank you for that 😁.

I also now understand what you mean with the `int` and `str` differences when dealing with `--fake-strain-seed`, I will perform these changes now.

Reply:
Without the `nargs='+'` argument for `--fake-strain-seed` I get the error `ValueError: invalid literal for int() with base 10: 'H'`.